dm/docs/template_dev_guide.md

```diff
@@ -101,6 +102,42 @@ Always run the test from the root of the `cloud-foundation` project:
 For the sake of consistency, keep the test files similar, as much as possible,
 to the *example configs* available in each template's `examples/` directory.
 
+
+### Running Bats tests with docker image
+
+You can use Developer Tools docker image with Bats and other needed tools installed. 
+Check cloud-foundation-toolkit/infra/concourse/build/developer-tools/Makefile for DOCKER_TAG_VERSION_DEVELOPER_TOOLS version.
+
+    export DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.2.0
+ 
+Create GCloud project and Service Account with permissions needed to run tests (they listed in README.md)
+Export Service Account as json file and export corresponding ENV variable, for example:
+
+    export SERVICE_ACCOUNT_JSON=$(< my-service-account.json)
+
+Go to cloud-foundation-toolkit/md folder
+    
+    cd cloud-foundation-toolkit/md
+
+#### Run Docker container
+
+If you followed [Using the Cloud Foundation Config File](#using-the-cloud-foundation-config-file), you can mount config file on docker run:
+
+    docker run -it -e SERVICE_ACCOUNT_JSON=${SERVICE_ACCOUNT_JSON} -v `pwd`:/workspace -v ~/.cloud-foundation-tests.conf:/root/.cloud-foundation-tests.conf cft/developer-tools:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} bash
```

Comment:
We need a solution NOT to use SA json-s. Would it be possible to use the underlying gclouds authenticated (user) permission?

Reply:
https://googlecloud.tips/tips/005-mounting-your-local-gcloud-credentials-in-docker/

Reply:
https://medium.com/google-cloud/how-to-test-google-cloud-services-locally-in-docker-d74196147841

Reply:
https://hub.docker.com/r/google/cloud-sdk/

It looks like there are multiple options, good to mention in the documentation.

The easiest is the first one:
`docker run -v ~/.config/gcloud:/root/.config/gcloud  google/cloud-sdk gcloud auth list`

The other option is to create a local volume and mount it ( this takes 1 extra step of authentication):
```
docker run -ti --name gcloud-config google/cloud-sdk gcloud auth login
docker run --rm -ti --volumes-from gcloud-config google/cloud-sdkgcloud auth list
```

Reply:
I got the idea, will do